src/index/IndexImpl.h

```diff
@@ -41,6 +41,7 @@
 #include "util/File.h"
 #include "util/Forward.h"
 #include "util/HashMap.h"
+#include "util/Iterators.h"
```

Comment:
The  `InputRangeTypeErased` is not in `Iterators.h` 
is this an unrelated include fix, or should you use a more precise include here?

Reply:
doublechecked this and InputRangeTypeErased is in Iterators.h, might be that this include is not really necessary but I guess if the file uses the type then it should be there?

Reply:
You are right, I was wrong. Sorry. I have so many of these ranges-tools in the meantime that I am mixing up the header infrastructure. leave it as is.

Reply:
I let myself to resolve this conversation to get it out of the view.